src/Validation.Common.Job/Validation/ValidationResponse.cs

```diff
@@ -0,0 +1,133 @@
+        /// </summary>
+        public static IValidationResponse Failed { get; } = new ValidationResponse(ValidationStatus.Failed);
+
+        /// <summary>
+        /// Create a new validation step response with the given status.
+        /// </summary>
+        /// <param name="status">The step's status.</param>
+        public ValidationResponse(ValidationStatus status)
+            : this(status, results: null, outputUrl: null)
+        {
+        }
+
+        /// <summary>
+        /// Create a new validation step response with the given status.
+        /// </summary>
+        /// <param name="status">The step's status.</param>
+        /// <param name="outputUrl">
+        /// A URL to modified package content. Must be null if status is not 
+        /// <see cref="ValidationStatus.Succeeded"/>.
+        /// </param>
+        public ValidationResponse(
+            ValidationStatus status,
+            string outputUrl)
+            : this(status, results: null, outputUrl: outputUrl)
+        {
+        }
+
+        /// <summary>
+        /// Create a new validation step repsonse with the given status.
+        /// </summary>
+        /// <param name="status">The step's status.</param>
+        /// <param name="results">
+        /// The results generated by the validation step. These provide additional information,
+        /// such as debugging information or reasons for failure.
+        /// </param>
+        public ValidationResponse(
+            ValidationStatus status,
+            IReadOnlyList<PackageValidationResult> results)
+            : this(status, results, outputUrl: null)
+        {
+        }
+
+        /// <summary>
+        /// Create a new failed validation step response with the given errors.
+        /// </summary>
+        /// <param name="status">The status of the validation step.</param>
+        /// <param name="results">
+        /// The results generated by the validation step. These provide additional information,
+        /// such as debugging information or reasons for failure.
+        /// </param>
+        /// <param name="outputUrl">
+        /// A URL to modified package content (.nupkg). Must be null if status is not 
+        /// <see cref="ValidationStatus.Succeeded"/>.
+        /// </param>
+        public ValidationResponse(
+            ValidationStatus status,
+            IReadOnlyList<PackageValidationResult> results,
+            string outputUrl)
+        {
+            if (results?.Count > 0 && status != ValidationStatus.Succeeded && status != ValidationStatus.Failed)
+            {
+                throw new ArgumentException("Cannot specify results if the validation is not in a terminal status.", nameof(status));
+            }
+
+            if(!string.IsNullOrEmpty(outputUrl) && status != ValidationStatus.Succeeded)
+            {
+                throw new ArgumentException("Cannot specify outputUrl if the validation is not in a success state.", nameof(status));
```

Comment:
Prefer using `nameof` so that this line will automatically get updated if we rename the variable.

```suggestion
                throw new ArgumentException($"Cannot specify {nameof(outputUrl)} if the validation is not in a success state.", nameof(status));
```